src/index.js

```diff
@@ -417,7 +418,15 @@ let Link = forwardRef(({ innerRef, ...props }, ref) => (
                 if (anchorProps.onClick) anchorProps.onClick(event);
                 if (shouldNavigate(event)) {
                   event.preventDefault();
-                  navigate(href, { state, replace });
+                  let shouldReplace = replace;
+                  if (replace !== "boolean" && isCurrent) {
```

Comment:
`typeof replace !== "boolean"` ?

Reply:
I only want to execute the is current replace feat if the user hasn't set the kind of `replace` she wants.

If I just evaluate `replace && isCurrent` and the user says she wants to replace and isCurrent is true, the if is going to evaluate to true, a `shallowCompare` will happen and `shouldReplace` could be false, even though the user is clearly specifying she wants to replace.

So, this feature just kick off if no `replace` options has been set.

Reply:
i think this is about the missing `typeof`

Reply:
Oh thanks @brendanmoore and @stefanprobst for pointing that out. I totally missed. What a silly mistake 😅.

I'll be creating a follow up PR to fix it.

Reply:
Ooof, great catch everyone! Should the unit tests not have caught this?

Reply:
I took care of this, thanks everyone for flagging!